src/js/components/form-elements/BaseField.jsx

```diff
@@ -1,31 +1,60 @@
-import React from 'react';
+import _ from 'lodash';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { Field } from 'redux-form';
 
 import { renderField } from '../../utils/form-utils';
 
-const BaseField = (props) => {
-  const {
-    fieldName,
-    fieldConfig,
-    renderInput,
-    touched,
-    arrayField,
-  } = props;
-  const dynamicAttr = fieldConfig.getDynamicAttr ? fieldConfig.getDynamicAttr(props) : {};
-
-  return (
-    <Field
-      name={fieldName}
-      component={renderField}
-      renderInput={renderInput}
-      attributes={{ ...fieldConfig.attributes, ...dynamicAttr }}
-      label={fieldConfig.label}
-      touched={touched}
-      arrayField={arrayField}
-    />
-  );
-};
+class BaseField extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      touched: false,
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.state.touched !== nextState.touched) {
```

Comment:
The touched property indicates if the element was visited (had focus), it's used in validation, the validation errors are displayed only when field is touched, so you don't have errors on every required field when displaying the form. By default React will update the component every time it will receive props even if those props was not changed. I'm using the shouldComponentUpdate method to limit rerenders and only do it when it's needed, because they are time consuming